packages/build-info/src/build-systems/package-managers.ts

```diff
@@ -0,0 +1,87 @@
+import { isNpmBuildScript, isNpmDevScript } from '../get-commands.js'
+import { PkgManager } from '../package-managers/detect-package-manager.js'
+
+import { BaseBuildTool, Command } from './build-system.js'
+
+export class PNPM extends BaseBuildTool {
+  id = 'pnpm'
+  name = 'PNPM'
+  runFromRoot = true
+
+  // will be called after the framework detection has run
+  async getCommands(packagePath: string): Promise<Command[]> {
+    const { name, scripts } = await this.project.fs.readJSON(
+      this.project.resolveFromPackage(packagePath, 'package.json'),
+    )
+    if (scripts && Object.keys(scripts).length > 0) {
+      return Object.entries(scripts).map(([scriptName, value]) => ({
+        type: isNpmDevScript(scriptName, value) ? 'dev' : isNpmBuildScript(scriptName, value) ? 'build' : 'unknown',
+        command: `pnpm run ${scriptName} --filter ${name}`,
+      }))
+    }
+    return []
+  }
+  async detect(): Promise<this | undefined> {
+    if (this.project.workspace && this.project.packageManager?.name === PkgManager.PNPM) {
+      return this
+    }
+  }
+}
+
+export class Yarn extends BaseBuildTool {
+  id = 'yarn'
+  name = 'Yarn'
+  runFromRoot = true
+
+  // will be called after the framework detection has run
+  async getCommands(packagePath: string): Promise<Command[]> {
+    const { name, scripts } = await this.project.fs.readJSON(
+      this.project.resolveFromPackage(packagePath, 'package.json'),
+    )
+    if (scripts && Object.keys(scripts).length > 0) {
+      return Object.entries(scripts).map(([scriptName, value]) => ({
+        type: isNpmDevScript(scriptName, value) ? 'dev' : isNpmBuildScript(scriptName, value) ? 'build' : 'unknown',
+        command: `yarn workspace ${name} ${scriptName}`,
+      }))
+    }
+    return []
+  }
+
+  async detect(): Promise<this | undefined> {
+    // only detect for yarn berry
```

Comment:
Could you elaborate on why we only detect berry?

Reply:
Addressed in https://github.com/netlify/build/pull/4977/commits/ac28711e57721593b35986c1b39480354cba73af